BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -966,6 +966,43 @@ public void ToggleRipple()
             craftPart.Dispose();
         }
 
+        [KSPField(isPersistant = true)]
+        public bool useThisWeaponForAim = false;
+
+        [KSPEvent(guiActive = false, guiActiveEditor = true, guiName = "#LOC_BDArmory_AimOverrideFalse")]//"Aim With This Weapon"
+        public void setAimOverride()
+        {
+            useThisWeaponForAim = !useThisWeaponForAim;
+            if (useThisWeaponForAim == false)
+            {
+                Events["setAimOverride"].guiName = StringUtils.Localize("#LOC_BDArmory_AimOverrideFalse");//"Aim With This Weapon"
+            }
+            else
+            {
+                Events["setAimOverride"].guiName = StringUtils.Localize("#LOC_BDArmory_AimOverrideTrue");//"Revert Aim Override"
+                List<Part>.Enumerator craftPart = EditorLogic.fetch.ship.parts.GetEnumerator();
+                while (craftPart.MoveNext())
+                {
+                    if (craftPart.Current == null) continue;
+                    List<ModuleWeapon>.Enumerator weapon = craftPart.Current.FindModulesImplementing<ModuleWeapon>().GetEnumerator();
+                    while (weapon.MoveNext())
+                    {
+                        if (weapon.Current == null) continue;
+                        if (weapon.Current == this) continue; //setting this here instead of craftPart.Current in case part has multiple weapon modules
+                        if (weapon.Current.GetShortName() != shortName) continue;
+                        if (weapon.Current.useThisWeaponForAim)
+                        {
+                            weapon.Current.useThisWeaponForAim = false;
+                            weapon.Current.Events["setAimOverride"].guiName = StringUtils.Localize("#LOC_BDArmory_AimOverrideFalse");//"Aim With This Weapon"
+                            GUIUtils.RefreshAssociatedWindows(weapon.Current.part);
+                        }
+                    }
+                    weapon.Dispose();
```

Comment:
Using `using ... .GetEnumerator();` without the `.Dispose();` is better practice, just in case something happens that prevents the `.Dispose()` from getting called.